File: verify_core.js

const { chromium } = require('playwright');

(async () => {
    console.log("Starting Core Verification...");
    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage();

    // Log console messages
    page.on('console', msg => console.log('BROWSER:', msg.text()));
    page.on('pageerror', err => console.log('BROWSER ERROR:', err));

    try {
        console.log("Navigating to localhost:5173...");
        await page.goto('http://localhost:5173', { timeout: 60000 });

        console.log("Waiting for canvas...");
        await page.waitForSelector('canvas', { timeout: 60000 });
        console.log("Canvas found. Game loaded.");

        // Wait to ensure physics is ticking
        await page.waitForTimeout(3000);

        console.log("Pressing Jump...");
        await page.keyboard.press('Space');

        await page.waitForTimeout(1000);
        console.log("Survival check passed.");

    } catch (e) {
        console.error("Verification Failed:", e);
        process.exit(1);
    } finally {
        await browser.close();
    }
})();

File: verify_core.py

from playwright.sync_api import sync_playwright

def verify_core():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        page.on("console", lambda msg: print(f"Console: {msg.text}"))
        try:
            page.goto("http://localhost:5173", timeout=10000)
            print(f"Title: {page.title()}")
            content = page.content()
            if "canvas" in content:
                print("Canvas found in HTML source")
            else:
                print("Canvas NOT found in HTML source")
        except Exception as e:
            print(f"Error: {e}")
        finally:
            browser.close()

if __name__ == "__main__":
    verify_core()
